extended/src/main/java/io/kubernetes/client/extended/controller/Controllers.java

```diff
@@ -50,6 +51,16 @@
    * @return the thread factory
    */
   public static ThreadFactory namedControllerThreadFactory(String controllerName) {
-    return new ThreadFactoryBuilder().setNameFormat(controllerName + "-%d").build();
+    final ThreadFactory defaultFactory = Executors.defaultThreadFactory();
+    final AtomicInteger threadNumber = new AtomicInteger(1);
+    String format = controllerName + "-%d";
+    return r -> {
+      Thread thread = defaultFactory.newThread(r);
+      if (!thread.isDaemon()) {
```

Comment:
Looking at the thread builder code, I don't think that we were previously setting threads to daemon.

Reply:
Guava was (it's the default). I can't think of any reason you'd want these to be non-daemon.

Reply:
I switched it back to non-daemon anyway. I don't know why they weren't daemon before but it does seem to work, so it's safer to leave it as is.